### PR TITLE
ADD: Documentation for SSP, Changelog and Puppet

### PR DESCRIPTION
**Pull request by brettt89**

### CODE REVIEW

File: docs/06_ssp_puppet.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+For any changes made to SSP, we should attempt to replicate by using Puppet as much as possible. this allows for easier customization of these services in the future.
+
+All puppet yaml configuration files can be found in `platform/config`. These files are split into the following
+
+ - apache.yaml
+ - nginx.yaml
+ - mysql.yaml
+ - common.yaml
+
+The first 3 yaml files are fairly obvious as to which configuraitons they are changing. All other non-major components have been defined in the `common.yaml` file (e.g. PHP, locales, sera, etc).
```
Comment (Firesphere) [inline docs/06_ssp_puppet.md]: A bit more on what the common.yaml contains would be good. It's a bit "You know, standard things and stuff" right now

Comment (brettt89) [reply docs/06_ssp_puppet.md]: Can you give a short example on what type of information you would like described?

Comment (Firesphere) [reply docs/06_ssp_puppet.md]: A quick explanation of each key-point mainly.

Like
```
base_includes:
- Default includes required for having the box represent the state of SSP correctly
php_extensions:
- Basic setup of the default extensions set up. If you want to add or remove any extensions you can add them here to modify your box to suit your needs
```
Etc.

Comment (brettt89) [reply docs/06_ssp_puppet.md]: base_includes and base_packages are listed below in "Puppet .yaml files" section as these can be used in any / all yaml files.

Will add a quick description for php_extensions

**Comment by brettt89** [comment]
@Firesphere Have updated docs to include section and examples regarding php_extensions

**Comment by Firesphere** [comment]
Yep, just a bit of clarification 👍

**Pull request merged**